lib/chef/knife/windows_cert_generate.rb

```diff
@@ -75,7 +76,7 @@ def prompt_for_passphrase
       end
 
       def generate_certificate rsa_key
-        @hostname = "*"
+        @hostname = Socket.gethostname
```

Comment:
Have we tried this? If it doesn't work for basic testing I'm not convinced it's a worthwhile default. What's the use case here, is someone going to run 'knife windows cert generate' then 'knife windows listener create' then 'knife bootstrap windows winrm localhost'?

That doesn't make sense, I maybe we're expecting the person to run 'knife windows cert generate' on the server that they're going to connect to? In that case, this just returns hostname, not fqdn. In my experience with HTTPS, you're almost always connecting to the fqdn. Is winrm any different? If we really want FQDN, we can't get it from Socket (without looking up reverse dns, which isn't what we want).